contracts/PangolinExchange.sol

```diff
@@ -48,18 +48,21 @@ contract PangolinExchange is Ownable, IAssetsExchange {
    * @dev _token ERC20 token's address
    * @dev _exactERC20AmountOut amount of the ERC20 token to be bought
   **/
-  function buyAsset(bytes32 _token, uint256 _exactERC20AmountOut) payable external override RefundRemainder {
+  function buyAsset(bytes32 _token, uint256 _exactERC20AmountOut) payable external override returns(bool){
     require(_exactERC20AmountOut > 0, "Amount of tokens to buy has to be greater than 0");
-
     address tokenAddress = supportedAssets.getAssetAddress(_token);
-
     uint256 amountIn = getEstimatedAVAXForERC20Token(_exactERC20AmountOut, tokenAddress);
-
     require(msg.value >= amountIn, "Not enough funds provided");
 
-    pangolinRouter.swapAVAXForExactTokens{value : msg.value}(_exactERC20AmountOut, getPathForAVAXtoToken(tokenAddress), msg.sender, block.timestamp);
+    address[] memory path = getPathForAVAXtoToken(tokenAddress);
+    (bool success,) = address(pangolinRouter).call{value : msg.value}(abi.encodeWithSignature("swapAVAXForExactTokens(uint256,address[],address,uint256)", _exactERC20AmountOut, path, msg.sender, block.timestamp));
 
+    refundAvaxBalance();
+    if (!success) {
```

Comment:
How about logging the purchase status
emit TokenPurchase(msg.sender, _exactERC20AmountOut, block.timestamp, success);
It will allow us to track the unsuccessful PangolinExchange operations separately
It will also make the code more compact 👍 

emit TokenPurchase(msg.sender, _exactERC20AmountOut, block.timestamp, success);
return success;